src/main/java/seedu/address/model/lesson/Lesson.java

```diff
@@ -0,0 +1,175 @@
+package seedu.address.model.lesson;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+
+/**
+ * Represents a Lesson in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public abstract class Lesson implements Comparable<Lesson> {
+    // Types of lesson
+    private static final String RECURRING = "Recurring Lesson";
+    private static final String MAKEUP = "Makeup Lesson";
+
+    // Time fields
+    private final Date date;
+    private final Time startTime;
+    private final Time endTime;
+    private final TimeRange timeRange;
+
+    // Data fields
+    private final Subject subject;
+    private final Set<Homework> homework = new HashSet<>();
+
+    /**
+     * Every field must be present and not null.
+     *
+     * @param date Date of lesson.
+     * @param startTime Start time of the lesson.
+     * @param endTime End time of the lesson.
+     * @param subject Subject of the lesson.
+     * @param homework Homework for the lesson.
+     */
+    public Lesson(Date date, Time startTime, Time endTime, Subject subject, Set<Homework> homework) {
```

Comment:
Lesson should take in `timeRange`. No need `startTime` and `endTime`

Reply:
Currently, I am using startTime to compare lessons so I figured I should include endTime in lesson as well even though it is not used (by my reasoning in the above comment). Do let me know what you think.